ocaml/typing/typecore.ml

```diff
@@ -5382,7 +5432,7 @@ and type_expect_
       let funct_mode, funct_expected_mode =
         match pm.apply_position with
         | Tail ->
-          let mode = Value.local_to_regional (Value.newvar ()) in
+          let mode, _ = Value.newvar_below (Value.max_with_regionality Regionality.regional) in
```

Comment:
Why the change here?

Reply:
We want `mode` to be a left-right mode, because it's later used to fill in the `mode_fun` of `Eliminated_optional_arg` etc. `mode_fun` needs to be left-right mode, because in `check_local_application_complete`, it's used on both sides of `submode`.

"meet X with regional" can be implemented as either `meet [X; regional]` or as `f X` where `f` is some left-morphism. The former would give a right mode, and the latter a left mode. Neither of them give a left-right mode.

So instead, we create a new mode variable.